src/main/java/seedu/address/model/item/ItemDate.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.item;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ItemDate {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Dates must follow the format yyyy-mm-dd or dd-mm-yyyy.";
+
+    // TODO: Handle validation
+    public static final String VALIDATION_REGEX = "yyyy-MM-dd";
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(VALIDATION_REGEX);
+
+    public static final String DATE_OUTPUT_PATTERN = "dd/MM/yyyy";
+
+    public final LocalDate itemDate;
+
+    public ItemDate(String unparsedDate) {
+        requireNonNull(unparsedDate);
+         checkArgument(isValidItemDatetime(unparsedDate), MESSAGE_CONSTRAINTS);
```

Comment:
Is there a space in front of checkArgument?

Reply:
Will remove it.